source/key.ts

```diff
@@ -0,0 +1,366 @@
+import * as pbkdf2 from 'pbkdf2';
+
+import { AES_CBC, AES_GCM, RSA_OAEP } from './algorithms';
+import {
+    isEdge,
+    convertArrayBufferViewToBase64,
+    convertArrayBufferViewToString,
+    convertArrayBufferToHexadecimal,
+    convertBase64ToArrayBufferView,
+    convertHexadecimalToArrayBuffer,
+} from './utils';
+
+const KEY_VERSION = 1;
+const ENCRYPT = 'encrypt';
+const DECRYPT = 'decrypt';
+
+enum D4LKeyTypes {
+    COMMON_KEY = 'ck',
+    DATA_KEY = 'dk',
+    MAIN_KEY = 'mk',
+    PASSWORD_KEY = 'pk',
+    ATTACHMENT_KEY = 'ak',
+    TAG_ENCRYPTION_KEY = 'tek',
+    APP_PRIVATE_KEY = 'apriv',
```

Comment:
here is another breaking change as I wanted to transform these values to an enum. it used to be:
```
    APP: {
        PRIVATE_KEY: 'apriv',
        PUBLIC_KEY: 'apub',
    },
    USER: {
        PRIVATE_KEY: 'upriv',
        PUBLIC_KEY: 'upub',
    },
```

Reply:
Your idea adds structure :+1: